subsys/partition_manager/CMakeLists.txt

```diff
@@ -4,6 +4,13 @@
 # SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
 #
 
+if (NOT CONFIG_FLASH_MAP_CUSTOM)
+  message(FATAL_ERROR "CONFIG_FLASH_MAP_CUSTOM must be set when" \
+    "CONFIG_PARTITION_MANAGER_ENABLED is set")
+endif()
+
+zephyr_sources(flash_map_partition_manager.c)
```

Comment:
This should only be done if `CONFIG_PARTITION_MANAGER_ENABLED` is set AFAIU

Reply:
If CONFIG_PARTITION_MANAGER_ENABLED is not set, building will abort according to the if above.